osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs

```diff
@@ -67,6 +68,33 @@ protected ModDifficultyAdjust()
             }
         }
 
+        public virtual int AdjustedSettingsCount
```

Comment:
Is there a reason this is virtual and not using the same reflection magic that `IMod.HasNonDefaultSettings` is? Is it for the catch "hard rock offsets" thingy or something?

Reply:
Hmm, I was just avoiding reflection here... but it turned out to be required for the other usage so maybe just bite the bullet and do the same here?

Reply:
I'd say so at this point. Not sure this is saving much.